### PR TITLE
Fix issues found by Ruff

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

**Comment by brianhelba** [comment]
Note, on f8855f29effe3cc963e2d1787d4d91e0ae9f3be1, I had to make some judgement calls as to whether the newly raised exception was sufficiently related to the surrounding exception. Feel free to comment if you disagree anywhere.

File: dandiapi/api/asset_paths.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def update_asset_paths(old_asset: Asset, new_asset: Asset, version: Version):
 @transaction.atomic
 def add_version_asset_paths(version: Version):
     """Add every asset from a version."""
-    print('\t Leaves...')
+    # Leaves
```
Comment (mvandenburgh) [inline dandiapi/api/asset_paths.py:243]: @AlmightyYakob is there value in printing this out? Should we replace this with a `logger` call?

Comment (brianhelba) [reply dandiapi/api/asset_paths.py:243]: I don't know. I'm happy to change it to a `debug` or `info` log if you want.

Comment (AlmightyYakob) [reply dandiapi/api/asset_paths.py:243]: I'm kind of surprised those print statements are still there. I think they were originally added when the asset paths stuff was new, to make sure things were working correctly. You can remove them.

**Review by mvandenburgh — APPROVED** [review]

**Pull request merged**